stock_picking_auto_create_lot/models/stock_picking.py

```diff
@@ -2,25 +2,45 @@
 # Copyright 2020 ACSONE SA/NV
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 from odoo import models
+from odoo.osv import expression
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
+    def _prepare_auto_lot_domain(self, picking):
+        """
+        Prepare the domain to search for stock.move.line records
+        that require automatic lot assignment.
+        """
+        domain = [
+            ("picking_id", "=", picking.id),
+            ("lot_id", "=", False),
+            ("lot_name", "=", False),
+            ("product_id.tracking", "!=", "none"),
+            ("product_id.auto_create_lot", "=", True),
+        ]
+        if not picking._check_immediate():
+            domain = expression.AND([domain, [("qty_done", ">", 0)]])
+        return domain
```

Comment:
Let's handle a recordset instead of just one picking.
 
```suggestion
    def _prepare_auto_lot_domain(self, immediate=False):
        """Prepare the domain to search for stock.move.line records that require
        automatic lot assignment.
        """
        domain = [
            ("picking_id", "in", self.ids),
            ("lot_id", "=", False),
            ("lot_name", "=", False),
            ("product_id.tracking", "!=", "none"),
            ("product_id.auto_create_lot", "=", True),
        ]
        if not immediate:
            domain = expression.AND([domain, [("qty_done", ">", 0)]])
        return domain
```